### ISSUE TITLE
[@types/ol] missing types in sub folders

### ISSUE THREAD

**Issue by yonzathuji** [DefinitelyTyped/DefinitelyTyped#44274]
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @hanreev 

If you do not mention the authors the issue will be ignored.

I'm using openlayers in typescript (angular 8).
Some classes are seem to be missing from ```index.d.ts```, making it impossible to auto import.

For example, 
```
new ZoomToExtent()
```
requires me to type 
```
import ZoomToExtent from 'ol/control/ZoomToExtent';
```
without the option for auto-completion.

When I added 
```
export { default as ZoomToExtent } from './control/ZoomToExtent'
```
to the ```index.d.ts``` the auto import worked.

Am I doing something wrong or does ```index.d.ts``` is missing some exports?

**Comment by hanreev** [comment]
not all classes are exported from [index.js](https://github.com/openlayers/openlayers/blob/master/src/ol/index.js), the typescript definition must reflect those exports or there will be a compilation error.

If you mean VSCode auto-import, AFAIK it will discover all members in a module after the first import.

**Comment by yonzathuji** [comment]
Do you know why not all classes are imported from `index.js`?

Maybe it's worth opening an issue there?

Not having all the classes automatically imported makes it pretty frustrating to work with.

**Comment by SimplyComple0x78** [comment]
Any updates on this? I have this issue, too, and can't get auto-import-suggestions in VSC on the first import. It's really annoying having to import files as
```ts
import TileLayer from 'ol/layer/Tile';
// or
import Projection from 'ol/proj/Projection';
```
manually each time I need a new ol-feature (like, using WMTSes from `ol/source/WMTS`, it's tilegrid from `ol/tilegrid/WMTS` and so on).

**Comment by hanreev** [comment]
how about importing module members from its parent like:
```typescript
import {Graticule, Tile as TileLayer, Vector as VectorLayer} from 'ol/layer';
import {Tile, Vector, WMTS, XYZ} from 'ol/source';
import {Projection} from 'ol/proj';
```

`Ctrl`+`Space` may help you populating the autocomplete

**Comment by SimplyComple0x78** [comment]
Yes, I can, but this comes with a lot of renaming.
`import TileLayer from 'ol/layer/Tile';` is a lot nicer than `import {Tile as TileLayer} from 'ol/layer';` as the first is proposed by the IDE (in case you already imported anything from there, as this thread is all about...) and the latter is not. Shouldn't all ways for importing be supported by DefinitelyTyped?

**Comment by Aziz-JH** [comment]
can be closed because there ship there own since 6.3.0 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64248
